ai_feedback/image_processing.py

```diff
@@ -114,25 +116,18 @@ def process_image(args, prompt: dict) -> tuple[str, str]:
                 "{context}", "```\n" + context + "\n```"
             )
         if prompt.get("include_image_size", False):
-            submission_image_paths = read_submission_images(OUTPUT_DIRECTORY, question)
-            submission_image_path = submission_image_paths[
-                0
-            ]  # Only consider one image per question
-            image = PILImage.open(submission_image_path)
+            # Only consider one image per question
+            image = PILImage.open(args.submission_image)
             message.content = message.content.replace(
                 "{image_size}", f"{image.width} by {image.height}"
             )
         if prompt.get("include_submission_image", False):
-            submission_image_paths = read_submission_images(OUTPUT_DIRECTORY, question)
-            submission_image_path = submission_image_paths[
-                0
-            ]  # Only consider one image per question
+            # Only consider one image per question
+            submission_image_path = args.submission_image
             message.images.append(Image(value=submission_image_path))
         if prompt.get("include_solution_image", False):
-            solution_image_paths = read_solution_images(OUTPUT_DIRECTORY, question)
-            solution_image_path = solution_image_paths[
-                0
-            ]  # Only consider one image per question
+            # Only consider one image per question
+            solution_image_path = args.submission_image
```

Comment:
Ah, this is interesting. You should actually include another argument `--solution-image` for the instructor to specify a path to a solution image file. It shouldn't be the same as `--submission-image` (which is what the student submits).